api/krusty/kustomizer.go

```diff
@@ -66,7 +66,8 @@ func (b *Kustomizer) Run(
 		ldr,
 		b.depProvider.GetFieldValidator(),
 		resmapFactory,
-		pLdr.NewLoader(b.options.PluginConfig, resmapFactory),
+		// The plugin configs are always located on disk, regardless of the fSys passed in
+		pLdr.NewLoader(b.options.PluginConfig, resmapFactory, filesys.MakeFsOnDisk()),
```

Comment:
I'm having to hardcode this in a few places now that the plugin loader respects the filesystem passed in. This one is an interesting example. If I use the `fSys` argument, the in-memory FS case of `TestPluginEnvironment` fails (as does one other test). I think this is why: 

`MakeEnhancedHarness` always sets up env vars that refer to paths in the real FS. Previously, `th.MakeOptionsPluginsEnabled()` in  `TestPluginEnvironment - confirmBehavior` function would create and embed a `PluginConfig` that captured the directory information from the real FS (because it used `konfig.EnabledPluginConfig`, which previously hardcoded `filesys.MakeFsOnDisk()`). That `PluginConfig` would then be used here in `Kustomizer` , which means `Kustomizer` is using a plugin loader referencing paths in the real FS even in the "in-memory" test case: that test case loads config and plugins from different file systems. After my changes, the env is still set just fine, but the test cannot prove it because it is looking for the plugin at paths that do not exist in the in-memory filesystem (they never did).

So one possible solution is to (continue to) hardcode plugin loading to the real FS, as I've done above.

Reply:
ok